File: lib/ts/models/src/linked-art/LinkedArtHasSymbolicContentMixin.ts
import {crm} from "@paradicms/vocabularies";
import {Literal} from "@rdfjs/types";
import {Memoize} from "typescript-memoize";
import {LinkedArtModelMixin} from "./LinkedArtModelMixin";

export abstract class LinkedArtHasSymbolicContentMixin extends LinkedArtModelMixin {
  @Memoize()
  get hasSymbolicContent(): readonly Literal[] {
    return this.filterAndMapObjects(crm.P190_has_symbolic_content, term =>
      term.termType === "Literal" ? (term as Literal) : null
    );
  }

  protected preMemoizeHasSymbolicContent(): void {
    this.hasSymbolicContent;
  }
}

File: lib/py/etl/paradicms_etl/models/linked_art/linked_art_has_symbolic_content_mixin.py
from rdflib import Literal

from paradicms_etl.models.linked_art.linked_art_model_mixin import LinkedArtModelMixin
from paradicms_etl.namespaces.crm import CRM


class LinkedArtHasSymbolicContentMixin(LinkedArtModelMixin):
    @property
    def has_symbolic_content(self) -> tuple[Literal, ...]:
        return tuple(
            self._values(CRM.P190_has_symbolic_content, self._map_term_to_literal)
        )
